intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/indexing/html_parser.py

```diff
@@ -0,0 +1,172 @@
+import requests
+from urllib.parse import urlparse, urlunparse
+import multiprocessing
+import urllib3
+import langid
+import PyPDF2
+from bs4 import BeautifulSoup
+import os
+import re
+from .context_utils import uni_pro
+
+
+urllib3.disable_warnings()
+
+class Crawler:
+    def __init__(self, pool=None):
+        if pool:
+            assert isinstance(pool, (str, list, tuple)), 'url pool should be str, list or tuple'
+        self.pool = pool
+        self.headers = {
+            'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7',

```

Comment:
is it browser specific?

Reply:
just confirmed with Guo Heng, this is just a simulator to imitate the behavior of a browser